src/common/system/proc_parser.cc

```diff
@@ -774,5 +774,77 @@ StatusOr<absl::flat_hash_set<std::string>> ProcParser::GetMapPaths(pid_t pid) co
   return map_paths;
 }
 
+StatusOr<absl::flat_hash_set<ProcParser::ProcessMap>> ProcParser::GetMapEntries(
+    pid_t pid, std::string libpath) const {
+  absl::flat_hash_set<ProcParser::ProcessMap> map_entries;
+
+  static constexpr int kProcMapNumFields = 6;
+
+  const std::filesystem::path proc_pid_maps_path = ProcPidPath(pid) / "maps";
+  PL_ASSIGN_OR_RETURN(std::string content, px::ReadFileToString(proc_pid_maps_path));
+  std::ifstream ifs;
+  ifs.open(proc_pid_maps_path);
+  if (!ifs) {
+    return error::Internal("Failed to open file $0", proc_pid_maps_path.string());
+  }
+
+  std::string line;
+  while (std::getline(ifs, line)) {
+    // We need to match the header lines which are of the following form:
+    // address                   perms offset   dev    inode             pathname
+    // For example:
+    // 55e816b37000-55e816b65000 r--p  00000000 103:02 55579316          /usr/bin/vim.basic
+    // We differentiate these headers from the subsequent key-value pairs that include
+    // data about the memory usage for each of the process mappings.
+    auto idx = line.find(':');
+    if (idx == std::string::npos) {
+      continue;
+    }
+    // If the character after the colon is whitespace, this is a key-value line.
+    // Else the colon is part of the device (major:minor) and this is a header.
+    // Perhaps we should look for other indicators?
```

Comment:
This looks like code we have in `ParseProcPIDSMaps()`. Seems like an opportunity to refactor and share?

Reply:
This was refactored in a032909130d1246d66d6fbb1adadabc6ee7b277e. I enhanced `ProcParser::ProcessSMaps` to work for my use case and consolidated the proc_parser function to work for both.